File: ProblemSet/41. 最大子数组/java/41. 最大子数组 贪心法.java
public class Solution {
    /**
     * @param nums: A list of integers
     * @return: A integer indicate the sum of max subarray
     */
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE, sum = 0;
        for (int num : nums) {
            sum += num;
            maxSum = Math.max(maxSum, sum);
            sum = Math.max(sum, 0);
        }
        return maxSum;
    }
}

File: ProblemSet/41. 最大子数组/python/41. 最大子数组 贪心法.py
import sys

class Solution:
    """
    @param nums: A list of integers
    @return: A integer indicate the sum of max subarray
    """
    def maxSubArray(self, nums):
        maxSum, cur = -sys.maxsize - 1, 0
        for num in nums:
            cur += num
            maxSum = max(maxSum, cur)
            cur = max(cur, 0) 
        return maxSum
